### PR TITLE
Add 404 redirect support.

### LINKED ISSUE #19: Laravel's 404 page bypasses demo-mode
When in demo mode and visiting a route that doesn't exist, the 404 is displayed. 
This demo-mode page should be displayed instead.

### PR DESCRIPTION
**Pull request by dmfj**
Fixes #19.

### CODE REVIEW

**Comment by dmfj** [comment]
Oh no! Hmm...

ss

**Comment by pawelmysior** [comment]
The Router fallback method was introduced in Laravel 5.5.5, so you should probably include that requirement in the composer.json file.

As for a 5.5.0 to 5.5.4 compatible code: the fallback method registers something similar to:

```
Route::get('{any}', 'Controller@method')->where('any', '.*');
```

The problem is that if you add this to the ServiceProvider, this route will take precedence over other routes. I wonder if there's a way to register a route as the last one.

**Comment by freekmurze** [comment]
The latest version of this package is already L5.5 only, so I don't mind the router fallback method being used.

File: tests/TestCase.php [hunk 1]
```diff
@@ -23,6 +23,8 @@ public function setUp()
 
         $this->setUpRoutes($this->app);
 
+        $this->registerFallbackRoute();
```
Comment (freekmurze) [inline tests/TestCase.php:26]: Do we really need to register this callback manually in the tests? I'm thinking that the `DemoModeServiceProvider` would already to care of this, no?

**Review by freekmurze — COMMENTED** [review]
Nice PR, I only have one question.

**Comment by pawelmysior** [comment]
Yeah, I know that it's a Laravel 5.5 package. But the fallback method was introduced in 5.5.5 specifically, so this will not work for some people that have 5.5.2 for example.

It's not a big issue, since you can just update your Laravel version to anything 5.5.5 or greater but I just thought it could be useful to make a note of that.

**Comment by freekmurze** [comment]
> Yeah, I know that it's a Laravel 5.5 package. But the fallback method was introduced in 5.5.5 specifically, so this will not work for some people that have 5.5.2 for example.

Ah ok, thanks for clarifying that. @dmfj could you update the requirements in `composer.json`?

File: tests/TestCase.php (hunk 1 above)
Comment (dmfj) [reply tests/TestCase.php:26]: You would think so. It appears that we do, however. (The suite fails without this - even though I have confirmed that the route is indeed registered in the service provider via the browser.)

Let me know if you have any ideas regarding this.

**Comment by dmfj** [comment]
@freekmurze Done.

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**